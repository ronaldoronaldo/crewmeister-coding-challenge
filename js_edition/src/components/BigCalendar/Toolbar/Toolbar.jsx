import React from "react";
import moment from "moment";
import { ButtonsWrapper, ToolbarContainer, ToolbarButton, Title } from "./Toolbar.style.jsx";

export default function Toolbar(toolbar) {
  const goToDayView = () => {
    toolbar.onView('day');
  }
  const goToWeekView = () => {
    toolbar.onView('week');
  }
  const goToMonthView = () => {
    toolbar.onView('month');
  }

  const goToBack = () => { toolbar.onNavigate('PREV'); };
  const goToNext = () => { toolbar.onNavigate('NEXT'); };
  const goToCurrent = () => { toolbar.onNavigate('TODAY'); };

  const renderTitle = () => {
    const date = moment(toolbar.date);
    return (
      <Title><b>{date.format('MMMM')}</b> {date.format('YYYY')}</Title>
    );
  };

  return (
    <ToolbarContainer>
      <ButtonsWrapper>
        <ToolbarButton onClick={goToCurrent}>Today</ToolbarButton>
        <ToolbarButton onClick={goToBack}>Back</ToolbarButton>
        <ToolbarButton onClick={goToNext}>Next</ToolbarButton>
      </ButtonsWrapper>
      {renderTitle()}
      <ButtonsWrapper>
        <ToolbarButton onClick={goToDayView}>Day</ToolbarButton>
        <ToolbarButton onClick={goToWeekView}>Week</ToolbarButton>
        <ToolbarButton onClick={goToMonthView}>Month</ToolbarButton>
      </ButtonsWrapper>
    </ToolbarContainer>
  );
}
