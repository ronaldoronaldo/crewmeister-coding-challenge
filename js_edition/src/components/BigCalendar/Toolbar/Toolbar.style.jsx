import styled from "styled-components";
import {colors, fontWeights, radius, screens, spacing, text, transitions} from "../../../config/styleGuide";

export const ToolbarStyle = styled.div`

`;
