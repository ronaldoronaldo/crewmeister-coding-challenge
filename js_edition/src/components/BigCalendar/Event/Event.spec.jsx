import React from "react";
import { render } from "@testing-library/react";
import moment from "moment";

import Toolbar from "./Event";

describe("<Event />", () => {
  it("should match snapshot when is sickness", () => {
    const event = {
      id: 1,
      title: "Mike",
      allDay: true,
      type: "sickness",
      start: moment("2017-01-01"),
      end: moment("2017-01-01"),
    };
    const { container } = render(<Toolbar event={event} />);
    expect(container).toMatchSnapshot();
  });
  it("should match snapshot when is on vacation", () => {
    const event = {
      id: 2,
      title: "Tom",
      allDay: true,
      type: "vacation",
      start: moment("2017-01-02"),
      end: moment("2017-01-04"),
    };
    const { container } = render(<Toolbar event={event} />);
    expect(container).toMatchSnapshot();
  });
});
