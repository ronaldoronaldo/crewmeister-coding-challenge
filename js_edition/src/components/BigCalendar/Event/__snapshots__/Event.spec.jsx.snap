// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> should match snapshot when is on vacation 1`] = `
<div>
  <div
    class="sc-AxjAm iqqVjM"
  >
    <span
      class="sc-AxirZ cOgLMt"
    >
      Tom
    </span>
    <span
      class="sc-AxirZ sc-AxiKw iLSsjr"
    >
       - 
      vacation
    </span>
  </div>
</div>
`;

exports[`<Event /> should match snapshot when is sickness 1`] = `
<div>
  <div
    class="sc-AxjAm eAGnuO"
  >
    <span
      class="sc-AxirZ cOgLMt"
    >
      Mike
    </span>
    <span
      class="sc-AxirZ sc-AxiKw iLSsjr"
    >
       - 
      sickness
    </span>
  </div>
</div>
`;
